dataset/create_twitter_dataset.py

```diff
@@ -0,0 +1,199 @@
+                #   4) A reads B's timeline (as an effect of retweeting and replying)
+                #   5) A reads A's timeline
+                access_timeline(timestamp, second_uid, first_user_ip, second_target_server, result_file, user_timeline_map, tweet_server, reply_stream_map)
+                access_timeline(timestamp, first_uid, first_user_ip, first_target_server, result_file, user_timeline_map, tweet_server, reply_stream_map)
+                result_file.write(LOG_FORMAT % (timestamp, tweet_uid, first_user_ip, 'null', first_target_server, 'WRITE', '201', '0'))
+                tweet_server[tweet_uid] = first_target_server  # must be here, because this is were the tweet is already stored on the server.
+                if first_uid not in user_timeline_map:  # add the tweet to the timeline
+                    user_timeline_map[first_uid] = []
+                elif len(user_timeline_map[first_uid]) > TIMELINE_LIMIT:
+                    user_timeline_map[first_uid].pop(0)  # remove the first element
+                user_timeline_map[first_uid].append(tweet_uid)  # add the tweet to the timeline
+                if mode == TWITTER_RE:
+                    reply_stream_map[tweet_uid] = second_uid # the tweet is directed to the second_uid
+
+                result_file.write(LOG_FORMAT % (timestamp, second_uid, first_user_ip, tweet_uid, second_target_server, 'READ', '200', str(TWEET_SIZE)))
+                access_timeline(timestamp, first_uid, first_user_ip, first_target_server, result_file, user_timeline_map, tweet_server, reply_stream_map)
+            uuid_file.write('%s\n' % tweet_uid)
+
+# Helper method for populating log entries for reading from a timeline
+#
+# params:
+#   timestamp: the timestamp
+#   timeline_uuid: the uuid of the source's target timeline
+#   source: the source's ip address
+#   target_server: the target server's ip address
+#   result_file: the file objet for writing the logs
+#   user_timeline_map: a mapping from user uuid --> timeline content
+#   tweet_server: a mapping from tweet uuid --> server
+#   reply_stream_map: a mapping from reply tweet uuid --> destination stream
+def access_timeline(timestamp, timeline_uuid, source, target_server, result_file, user_timeline_map, tweet_server, reply_stream_map):
+    # Read source's timeline.
+    timeline_list = [] if timeline_uuid not in user_timeline_map else user_timeline_map[timeline_uuid] # get the list containing the uuid of the tweets for that timeline
+    read_size = TWEET_SIZE * len(timeline_list) if len(timeline_list) > 0 else TWEET_SIZE
+    result_file.write(LOG_FORMAT % (timestamp, timeline_uuid, source, 'null', target_server, 'READ', '200', str(read_size)))
+    # Read each tweet from the server.
+    for tweet in timeline_list:
+        read_target_server = tweet_server[tweet]
+        result_file.write(LOG_FORMAT % (timestamp, tweet, source, timeline_uuid, read_target_server, 'READ', '200', str(TWEET_SIZE)))
```

Comment:
Like above note, not sure if it should be `source` here or `target_server` as the source for these requests